hagrid/reservations/export_pdf.py

```diff
@@ -125,18 +125,33 @@ def generate_qr_code(link: str):
     return qr.make_image()
 
 
-def generate_collection_list(reservation: Reservation):
-    variation_dict = {}
-    for res_part in ReservationPart.objects.all().filter(reservation=reservation):
-        for p in ReservationPosition.objects.all().filter(part=res_part):
-            if p.variation in variation_dict:
-                variation_dict[p.variation] += p.amount
-            else:
-                variation_dict[p.variation] = p.amount
-    article_list = []
-    for entry in variation_dict:
-        article_list.append((entry, variation_dict[entry]))
-    return sorted(article_list, key=lambda t: str(t[0]))
+def generate_collection_list(reservation: Reservation, distinct_required = False):
+    groups = []
+    titles = []
+
+    part_set = []
+
+    if distinct_required:
+        for part in ReservationPart.objects.all().filter(reservation=reservation):
+            part_set.append([part])
+            titles.append("packing list for part {0}:".format(str(part.title)))
+    else:
+        part_set = [ReservationPart.objects.all().filter(reservation=reservation)]
+        titles.append("packing list:")
+    for part in part_set:
+        variation_dict = {}
```

Comment:
Maybe use a `collections.Counter`? :)

Reply:
In my opinion this would result in some unnecessary overhead. We wouldn't really gain any convenience within the code but the frequent lookup within the counter are quite heavy on the resource side. After all we're just matching the titles by their id which is implemented in a efficient way using array indices but not using dictionary look ups.

If you really insist on using a counter I can manage that dough. So it is up to you.

Reply:
I guess you got me wrong. I want you to use a Counter instead of a plain dict for `variation_dict`. 
That counter is just a subclass of python dictionary. With it we can avoid the if-else-check in line 146 :)
So no overhead, but cleaner code.